### ISSUE TITLE
SC2016 false positive with OpenShift client

### ISSUE THREAD

**Issue by dereckson** [koalaman/shellcheck#3033]
This issue is very similar to #1161, but when the container is run with `oc rsh` instead of `docker`.

`oc` is the OpenShift client, and `oc rsh` allows to get a remote shell in a pod container.

#### For bugs
- Rule Id (if any, e.g. SC1000): SC2016
- My shellcheck version: 0.10.0
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
oc rsh "$pod" sh -c 'mysql -uroot -p$MARIADB_ROOT_PASSWORD -e "SELECT 1+1;"'
```

#### Here's what shellcheck currently says:

```
In test-SC2016-oc.sh line 476:
oc rsh "$pod" sh -c 'mysql -uroot -p$MARIADB_ROOT_PASSWORD -e "SELECT 1+1;"'
                    ^-- SC2016 (info): Expressions don't expand in single quotes, use double quotes for that.
```

#### Here's what I wanted or expected to see:

It passes, like for sudo and docker commands.
